### ISSUE TITLE
Ignore Pods in Terminating State

### ISSUE THREAD

**Issue by logbon72** [kubeshop/botkube#249]
**Is your feature request related to a problem? Please describe.**
There is a known issue in Kubernetes where liveness probe fails in Pods that are being terminated, i.e. in Terminating state. We noticed that for every new deployment we rollout, we get error alerts about Pod liveness state, which is due to the Kubernetes bug referenced.

See Kubernetes issue: https://github.com/kubernetes/kubernetes/issues/52817

**Describe the solution you'd like**
Filter out errors from pods. in terminating state, or allow a configuration to do this.

**Additional context**
Kubernetes Version is v1.15.6, BotKube version is 0.9.1

**Comment by semoac** [comment]
🙏

**Comment by PrasadG193** [comment]
We should write a filter to check in case if error events, check if the status is `Terminating`, ignore skip the event.

**Comment by codenio** [comment]
I would like to take this up, kindly assign it to me.

**Comment by codenio** [comment]
@logbon72 can you provide a dummy YAML of the pod in Terminating state masking sensitive information

**Comment by PrasadG193** [comment]
@A-kanksh-a Are you working on this?

**Comment by A-kanksh-a** [comment]
@PrasadG193 No, I have not done any conclusive work on this issue.

**Comment by codenio** [comment]
can someone help me by providing pod describe YAML, in this case?
or steps to replicate this behavior.?

**Comment by rkennedy-tpl** [comment]
I believe I observed this. pods terminating throw liveness-probe and readiness-probe errors via BotKube and it's somewhat "unexpected" because the deployment is proceeding as expected.

so I went to hunt for an example of the Pod YAML during termination....

> Note: When a Pod is being deleted, it is shown as Terminating by some kubectl commands. This Terminating status is not one of the Pod phases. A Pod is granted a term to terminate gracefully, which defaults to 30 seconds. You can use the flag --force to terminate a Pod by force.

from the bottom of k8s doc section => https://kubernetes.io/docs/concepts/workloads/pods/pod-lifecycle/#pod-phase

there may not be a good signal to detect this behavior.

I observed that `phase` never changed from `Running`. the `containerStatuses` structure eventually shows a `state: terminated` but shortly after, the pod is deleted and the `get pod` now returns "not found". I don't think there is a "Status: Terminating" 😞 

```
$ kubectl get pod podname -n namespace -o yaml
...
status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2021-04-13T23:30:52Z"
    status: "True"
    type: Initialized
  - lastProbeTime: null
    lastTransitionTime: "2021-04-14T18:41:44Z"
    message: 'containers with unready status: [api]'
    reason: ContainersNotReady
    status: "False"
    type: Ready
  - lastProbeTime: null
    lastTransitionTime: "2021-04-14T18:41:44Z"
    message: 'containers with unready status: [api]'
    reason: ContainersNotReady
    status: "False"
    type: ContainersReady
  - lastProbeTime: null
    lastTransitionTime: "2021-04-13T23:30:52Z"
    status: "True"
    type: PodScheduled
  containerStatuses:
  - containerID: containerd://317825e10b35afe0c55f24c8c031183a604ccf71ab5602fff397352742bb39b6
    image: 1234567890.dkr.ecr.us-east-1.amazonaws.com/containerimage:202101010000
    imageID: 1234567890.dkr.ecr.us-east-1.amazonaws.com/containerimage@sha256:8cd46d1d6816a9a31118375dc654b6be74d657af2b8852d7e2e69a3f8165eb6a
    lastState: {}
    name: api
    ready: false
    restartCount: 0
    started: false
    state:  
      terminated:
        containerID: containerd://317825e10b35afe0c55f24c8c031183a604ccf71ab5602fff397352742bb39b6
        exitCode: 137
        finishedAt: "2021-04-14T18:42:19Z"
        reason: Error
        startedAt: "2021-04-13T23:34:01Z"
  hostIP: 10.123.94.20
  phase: Running
  ...
  ...
  ```

**Comment by mszostok** [comment]
I was not able to reproduce the described issue. The mentioned issue (https://github.com/kubernetes/kubernetes/issues/52817) was also resolved for K8s since 1.18. 

Additionally, We will follow the same support strategy as K8s does. 
> The Kubernetes project maintains release branches for the most recent three minor releases (1.24, 1.23, 1.22)

Feel free to reopen it if you will still encounter this problem with the newest version.

**Comment by akshayparseja** [comment]
my whole channel is filled with livenessprobe failure on pod deploymeny, can we ignore those?
